### PR TITLE
fix: missing withContext in simp trace

### PR DESCRIPTION
**Pull request by digama0**
As [reported on Zulip](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/simp.3F.20.5B*.5D/near/322724789).

### CODE REVIEW

File: src/Lean/Elab/Tactic/Simp.lean [hunk 1]
```diff
@@ -265,7 +265,7 @@ def traceSimpCall (stx : Syntax) (usedSimps : UsedSimps) : MetaM Unit := do
     | .fvar fvarId => -- local hypotheses in the context
       if let some ldecl := lctx.find? fvarId then
         localsOrStar := localsOrStar.bind fun locals =>
-          if !ldecl.userName.isInaccessibleUserName &&
+          if !ldecl.userName.isInaccessibleUserName && !ldecl.userName.hasMacroScopes &&
```
Comment (Kha) [inline src/Lean/Elab/Tactic/Simp.lean:268]: This seems like an unrelated change, does it affect the same test case? Should `isInaccessibleUserName` even be relevant here?

Comment (digama0) [reply src/Lean/Elab/Tactic/Simp.lean:268]: I think it is indeed an unrelated bugfix. I believe this code is chosen to match `Name.toString`, since this is what is used to ultimately print the local names and it should be re-parseable, but I agree that it would have to be pretty weird code for the `isInaccessibleUserName` conjunct to trigger.

Comment (Kha) [reply src/Lean/Elab/Tactic/Simp.lean:268]: Ok, if you could add a first test with a named hypothesis and then a second commit replacing `isInaccessibleUserName` with `hasMacroScopes` with the unnamed test case, that would be great.

Comment (Kha) [reply src/Lean/Elab/Tactic/Simp.lean:268]: I.e. separating the context and inaccessible fix

Comment (semorrison) [reply src/Lean/Elab/Tactic/Simp.lean:286]: I've separated this into two commits, and verified both changes are needed.
